src/toolkitAPI.c

```diff
@@ -815,6 +821,685 @@ int DLLEXPORT swmm_getSubcatchOutConnection(int index, int *type, int *out_index
+// Input:   index = Index of desired subcatchment 
+// Output:  int = number of lid units for subcatchment 
+// Return:  number of lid units for subcatchment
+// Purpose: count number of lid units for subcatchment
+{
+    int errcode = 0;
+    
+    // Check if Open
+    if(swmm_IsOpenFlag() == FALSE)
+    {
+        errcode = ERR_API_INPUTNOTOPEN;
+    }
+    // Check if subcatchment index is within bounds
+    else if(index < 0 || index >= Nobjects[SUBCATCH])
+    {
+        errcode = ERR_API_OBJECT_INDEX;
+    }
+    else
+    {
+        *value = lid_getLidUnitCount(index);
+    }
+    
+    return (errcode);
+}
+
+
+int DLLEXPORT swmm_getLidUParam(int index, int lidIndex, int param, double *value)
+//
+// Input:   index = Index of desired subcatchment 
+//          lidIndex = Index of desired lid unit (subcatchment allow for multiple lid units)
+//          param = Parameter desired (Based on enum SM_LidUProperty)
+// Output:  value = value to be output
+// Return:  API Error
+// Purpose: Gets Lid Unit Parameter
+{
+    int errcode = 0;
+    TLidUnit* lidUnit;
+
+    // Check if Open
+    if(swmm_IsOpenFlag() == FALSE)
+    {
+        errcode = ERR_API_INPUTNOTOPEN;
+    }
+    // Check if subcatchment index is within bounds
+    else if(index < 0 || index >= Nobjects[SUBCATCH])
+    {
+        errcode = ERR_API_OBJECT_INDEX;
+    }
+    else
+    {
+        lidUnit = lid_getLidUnit(index, lidIndex, &errcode);
+
+        // There are no Lid Units defined for the subcatchments
+        if(lidUnit)
+        {
+            switch(param)
+            {
+                case SM_UNITAREA:
+                    *value = lidUnit->area * SQR(UCF(LENGTH)); 
+                    break;
+                case SM_FWIDTH:
+                    *value = lidUnit->fullWidth * UCF(LENGTH); break;
+                case SM_BWIDTH:
+                    *value = lidUnit->botWidth * UCF(LENGTH); break;
+                case SM_INITSAT:
+                    *value = lidUnit->initSat * 100; break;
+                case SM_FROMIMPERV:
+                    *value = lidUnit->fromImperv * 100; break;
+                case SM_FROMPERV:
+                    *value = lidUnit->fromPerv * 100; break;
+                default:
+                    errcode = ERR_API_OUTBOUNDS; break;
+            }
+        }
+    }
+
+    return(errcode);
+}
+
+
+int DLLEXPORT swmm_setLidUParam(int index, int lidIndex, int Param, double value)
+//
+// Input:   index = Index of desired subcatchment 
+//          lidIndex = Index of desired lid unit (subcatchment allow for multiple lid units)
+//          param = Parameter desired (Based on enum SM_LidUProperty)
+//          value = replacement
+// Return:  API Error
+// Purpose: Gets Lid Unit Parameter
+{
+    int errcode = 0;
+    TLidUnit* lidUnit;
+
+    // Check if Open
+    if(swmm_IsOpenFlag() == FALSE)
+    {
+        errcode = ERR_API_INPUTNOTOPEN;
+    }
+    // Check if subcatchment index is within bounds
```

Comment:
@jennwuu,

I think you should limit the time to `set` parameters to before starting the simulation.  I suggest you add this:

```C
    // Check if Open
    if(swmm_IsOpenFlag() == FALSE)	
    {	  
        errcode = ERR_API_INPUTNOTOPEN;	 
    }	 
    // Check if Simulation is Running	
    else if(swmm_IsStartedFlag() == TRUE)	  
    {	  
        errcode = ERR_API_SIM_NRUNNING;	     
    }	  
    else	   
```